### PR TITLE
Dx improvements

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

**Comment by greg0ire** [comment]
> Nice, does it work on MacOS now?

You tell me, I do not use that operating system :P

File: docker/php-console/entrypoint.sh [hunk 1]
```diff
@@ -3,13 +3,4 @@ set -e
...
 setup
 
-# Wait for composer vendor
-if [ ! -f vendor/autoload.php ]; then
-    echo "Run 'composer install' command to continue."
-fi
-while [ ! -f vendor/autoload.php ]; do
-    sleep 1
-done
-echo "Composer vendor found, processing."
```
Comment (Soullivaneuh) [inline docker/php-console/entrypoint.sh]: Why removing this part?

Comment (greg0ire) [reply docker/php-console/entrypoint.sh]: Why would we wait before running `tail -f`? I think this was useful before, to tell people they need to run `composer` but now it works automatically

File: docker-compose.yml [hunk 2]
```diff
@@ -31,5 +34,15 @@ console:
...
-  env_file:
-    - docker-compose.env
+  environment:
+    - UNIX_USERNAME
+    - UNIX_GROUP
+    - UNIX_UID
+    - UNIX_GID
+
+composer:
+  image: composer
```
Comment (Soullivaneuh) [inline docker-compose.yml]: We already have composer and plugins we need on the php-dev image.

Comment (greg0ire) [reply docker-compose.yml]: Yeah but how are you supposed to run `composer install`? It's not documented…

Comment (Soullivaneuh) [reply docker-compose.yml]: `"echo "Run 'composer install' command to continue."`

On the script you deleted. :-)

I agree it's not very efficient, remember I did it very quickly at start. #DevExcuse :stuck_out_tongue:

File: docker/php-console/entrypoint.sh [hunk 3]
```diff
@@ -3,13 +3,8 @@ set -e
 
 setup
 
-# Wait for composer vendor
 if [ ! -f vendor/autoload.php ]; then
-    echo "Run 'composer install' command to continue."
+	su "$(stat -c '%U' composer.json)" -c 'composer install'
```
Comment (greg0ire) [inline docker/php-console/entrypoint.sh]: @Soullivaneuh I reimplemented this, this time using the composer binary from the php-console image. Looks good?

File: docker-compose.yml (hunk 2 above)
Comment (greg0ire) [reply docker-compose.yml]: Fixed

File: docker/php-console/entrypoint.sh [hunk 4]
```diff
@@ -3,13 +3,8 @@ set -e
 
 setup
 
-# Wait for composer vendor
 if [ ! -f vendor/autoload.php ]; then
-    echo "Run 'composer install' command to continue."
+	su "$UNIX_USERNAME" -c 'composer install'
```
Comment (greg0ire) [inline docker/php-console/entrypoint.sh]: @Soullivaneuh I reimplemented this, this time using the composer binary from the php-console image. Looks good?

Comment (Soullivaneuh) [reply docker/php-console/entrypoint.sh]: Let's try like that.

**Pull request merged**